src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(value);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<T>(tail, value);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else {
+            indexExist(index);
+            Node<T> current = nodeGetByIndex(index);
+            Node<T> newNode = new Node<T>(current.prev, value, current);
+            if (current == head) {
+                head = newNode;
+            } else {
+                current.prev.next = newNode;
+                current.prev = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexExist(index);
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        return currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexExist(index);
+        Node<T> currentNode = nodeGetByIndex(index);
+        T old = currentNode.value;
+        currentNode.value = value;
+        return old;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexExist(index);
+        Node<T> currentNode = nodeGetByIndex(index);
+        unlink(currentNode);
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (currentNode.value == object
+                    || currentNode.value != null && currentNode.value.equals(object)) {
+                unlink(currentNode);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void indexExist(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
```

Comment:
include message with `index` and `size`, to make debugging for users of your collection easier